### PR TITLE
[Fix/#60632] - Added get_real_file to decrypt the json file

### LINKED ISSUE #60632: uri: sourcing vaulted file doesn't work
##### SUMMARY

I'm using `uri` module to send the vaulted file, but Ansible doesn't decrypt it.

```
    uri:
      url: "http://localhost:8888/"
      method: POST
      src: file-encrypted.json
```

Instead, it's sending raw vault format, despite specifying `--ask-vault-pass`.

##### ISSUE TYPE
- ~Bug Report~
- Feature Idea

##### COMPONENT NAME

uri

##### ANSIBLE VERSION

```paste below
ansible 2.8.0
```

##### CONFIGURATION

n/a

##### OS / ENVIRONMENT

macOS

##### STEPS TO REPRODUCE

1. Create a `file.json` with `{'foo': 'bar'}` content.
2. Vault it via: `ansible-vault encrypt file.json`
3. Create the following playbook (e.g. `uri.yml`):

```yaml
---
- hosts: localhost
  tasks:
  - name: Send JSON file
    uri:
      url: "http://localhost:8888/"
      method: POST
      src: file.json
      headers:
        Content-Type: "application/json"
```
4. On the other terminal, run the netcat on port 8888: `nc -vl 8888`.
5. Run playbook via: `ansible-playbook uri.yml --ask-vault-pass -vvv`

##### EXPECTED RESULTS

Original content of JSON file is sent.

##### ACTUAL RESULTS

Encrypted version is received instead:

```paste below
$  nc -vl 8888
POST / HTTP/1.1
Accept-Encoding: identity
Host: localhost:8888
User-Agent: ansible-httpget
Content-Type: application/json
Content-Length: 355
Connection: close

$ANSIBLE_VAULT;1.1;AES256
39373739353265643239323865303463303638656538663366663330343835353834306462646232
3332326234303438396263663337646134393563333165640a616433366364666161366231326463
```

### PR DESCRIPTION
**Pull request by hybridx**
##### SUMMARY
Fixes #60632 Added get_real_file to decrypt the JSON file which was going as encrypted

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ansible/lib/ansible/modules/net_tools/basics

##### ADDITIONAL INFORMATION
```paste below
https://github.com/ansible/ansible/issues/60632#issue-481119072
```

### CODE REVIEW

File: lib/ansible/plugins/action/uri.py [hunk 1]
```diff
@@ -46,6 +46,11 @@ def run(self, tmp=None, task_vars=None):
             self._transfer_file(src, tmp_src)
             self._fixup_perms2((self._connection._shell.tmpdir, tmp_src))
 
+            try:
+                decrypt = boolean(self._task.args.get('decrypt', True), strict=False)
+                tmp_src = self._loader.get_real_file(tmp_src, decrypt=decrypt)
+            except AnsibleFileNotFound as e:
+                raise AnsibleActionFail("could not find src=%s, %s" % (tmp_src, to_text(e)))
```
Comment (bcoca) [inline lib/ansible/plugins/action/uri.py]: like in copy and other action plugins that support this, you should add a toggle for the user to be able to actually post a vaulted file if that is what they want.

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/action/uri.py (hunk 1 above)
Comment (hybridx) [reply lib/ansible/plugins/action/uri.py]: Will do that and send the PR once done

Comment (hybridx) [reply lib/ansible/plugins/action/uri.py]: Screenshot from 2019-09-01 14-14-13

Comment (hybridx) [reply lib/ansible/plugins/action/uri.py]: Decrypt default is set to False

**Comment by hybridx** [comment]
@bcoca can you please review the code?

File: lib/ansible/modules/net_tools/basics/uri.py [hunk 2]
```diff
@@ -42,6 +42,13 @@
       - A password for the module to use for Digest, Basic or WSSE authentication.
     type: str
     aliases: [ password ]
+  decrypt:
+    description:
+      - Decrypts vaulted file with provided with appropriate password
+    type: bool
+    required: false
+    default: no
+    version_added: "2.10"
```
Comment (sivel) [inline lib/ansible/modules/net_tools/basics/uri.py]: To maintain consistency, keep just the `decrypt` header and `version_added: "2.10"` here and remove the rest.

Then add:

```
extends_documentation_fragment:
- decrypt
```

**Review by sivel — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/net_tools/basics/uri.py (hunk 2 above)
Comment (hybridx) [reply lib/ansible/modules/net_tools/basics/uri.py]: Hi I have added the document as requested
```  
decrypt:
    description:
      - Decrypts vaulted file with provided with appropriate password
    extends_documentation_fragment:
      - decrypt
    version_added: "2.10"
```

Comment (sivel) [reply lib/ansible/modules/net_tools/basics/uri.py]: This isn't what I requested.

**Review by sivel — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/net_tools/basics/uri.py (hunk 2 above)
Comment (hybridx) [reply lib/ansible/modules/net_tools/basics/uri.py]: @sivel  I'll work on this and update

Comment (hybridx) [reply lib/ansible/modules/net_tools/basics/uri.py]: @sivel if I remove **default** from the documentation the build fails in shippable testing

Comment (hybridx) [reply lib/ansible/modules/net_tools/basics/uri.py]: @sivel does this look good to you? You can let me know and we can change accordingly.

**Comment by hybridx** [comment]
Hey @sivel could you please review?

**Comment by hybridx** [comment]
@bcoca @sivel can you please review the code?

**Comment by bcoca** [comment]
@hybridx can you resolve the conflicts and CI issues?

waiting_on_contributor

**Pull request closed without merging**